### ISSUE TITLE
Bare Metal bootstrap fails with OVN

### ISSUE THREAD

**Issue by jdhirst1** [openshift/okd#47]
When testing baremetal installation with OVN, bootstrap keeps failing at the following point:

```
Jan 28 18:16:57 bootstrap.okd.example.net bootkube.sh[21846]: "99_openshift-machineconfig_99-worker-ssh.yaml": unable to get REST mapping for "99_openshift-machineconfig_99-worker-ssh.yaml": no matches for kind "MachineConfig" in version "machineconfiguration.openshift.io/v1"
Jan 28 18:16:57 bootstrap.okd.example.net bootkube.sh[21846]: [#3181] failed to create some manifests:
Jan 28 18:16:57 bootstrap.okd.example.net bootkube.sh[21846]: "99_openshift-machineconfig_99-master-ssh.yaml": unable to get REST mapping for "99_openshift-machineconfig_99-master-ssh.yaml": no matches for kind "MachineConfig" in version "machineconfiguration.openshift.io/v1"
Jan 28 18:16:57 bootstrap.okd.example.net bootkube.sh[21846]: "99_openshift-machineconfig_99-worker-ssh.yaml": unable to get REST mapping for "99_openshift-machineconfig_99-worker-ssh.yaml": no matches for kind "MachineConfig" in version "machineconfiguration.openshift.io/v1"
```

It seems that OVN keeps getting stuck here: 
```
Jan 28 18:10:13 master-0.okd.example.net hyperkube[6711]: time="2020-01-28T18:08:52Z" level=debug msg="exec(630): /usr/bin/ovs-ofctl --no-stats --no-names dump-flows br-int table=41,ip,nw_src=10.10.0.0/16"
Jan 28 18:10:13 master-0.okd.example.net hyperkube[6711]: time="2020-01-28T18:08:52Z" level=fatal msg="Timeout error while obtaining addresses for k8s-master-0.okd.example.net (timed out waiting for the condition)"

```

When it says it is obtaining address for k8s-master-0.okd.example.net, is that a DNS FQDN?

Networking Config:
```
networking:
  clusterNetwork:
  - cidr: 10.10.0.0/16
    hostPrefix: 23
  networkType: OVNKubernetes
  serviceNetwork: 
  - 10.11.0.0/16
```

Saw some interesting logs in ovnkube-master container:
```
time="2020-01-28T17:47:25Z" level=error msg="macAddress annotation not found for node \"master-0.okd.example.net\" "
time="2020-01-28T17:47:25Z" level=error msg="k8s.ovn.org/l3-gateway-config annotation not found for node \"master-0.okd.example.net\""
time="2020-01-28T17:47:25Z" level=info msg="Allocated node master-2.okd.example.net HostSubnet 10.129.0.0/23"
time="2020-01-28T17:47:25Z" level=info msg="Setting annotations map[k8s.ovn.org/node-subnets:{\"default\":\"10.129.0.0/23\"} ovn_host_subnet:<nil>] on node master-2.okd.example.net"
```

ovs-daemons logs:
```
2020-01-28T18:24:10.356Z|00043|jsonrpc|WARN|unix#109: receive error: Connection reset by peer
2020-01-28T18:24:10.357Z|00044|reconnect|WARN|unix#109: connection dropped (Connection reset by peer)
```
nbdb / sbdb logs:
```2020-01-28T18:41:52.781144564+00:00 stderr F 2020-01-28T18:41:52Z|00994|stream_ssl|WARN|SSL_accept: unexpected SSL connection close
2020-01-28T18:41:52.781278548+00:00 stderr F 2020-01-28T18:41:52Z|00995|jsonrpc|WARN|ssl:172.16.20.20:44506: receive error: Protocol error
2020-01-28T18:41:52.781522487+00:00 stderr F 2020-01-28T18:41:52Z|00996|reconnect|WARN|ssl:172.16.20.20:44506: connection dropped (Protocol error)
```

Ran a capture on master-0, I am seeing a bunch of retransmissions to bootstrap node (172.16.20.29), pcap attached:
[master-0.pcap.zip](https://github.com/openshift/okd/files/4124110/master-0.pcap.zip)

**Comment by vrutkovs** [comment]
Its odd that its `k8s-master-0.okd.example.net` in OVN pod although the hostname is `master-0.okd.example.net`

cc @openshift/openshift-team-sdn

**Comment by danwinship** [comment]
> When it says it is obtaining address for k8s-master-0.okd.example.net, is that a DNS FQDN?

No. There are several different per-node objects created in the OVN databases, with different prefixes before the node name to differentiate them.

**Comment by dmc5179** [comment]
I just ran into this issue. I have to do a "bare metal" install in a private AWS region for "reasons". The interesting part is that I've been installing all day. I've hit different issues here and there that I have to work around due to the region. I've had the bootstrap node come up 50 times today, no problem, just spinning around waiting for etcd nodes. And then all of a sudden, 20 minutes ago I hit this. In some AWS private regions it can take over an hour to recycle an IP. As in, I attached an IP to a host, didn't like the outcome, terminated, and made a new instance with the IP.

You're probably not in an AWS private region but that fact that this was working all day, 4.3.0, until just now, is interesting. I have scripts for generating to manifests and ignition files so I know I'm getting the same config files every time.
